react/src/components/UserProfileSettingModal.tsx

```diff
@@ -0,0 +1,304 @@
+      access_key: matchLoggedAccount?.access_key,
+      secret_key: matchLoggedAccount?.secret_key,
+    });
+  }
+
+  const { value, dispatchEvent } = useWebComponentInfo();
+  let parsedValue: {
+    isOpen: boolean;
+  };
+  try {
+    parsedValue = JSON.parse(value || "");
+  } catch (error) {
+    parsedValue = {
+      isOpen: false,
+    };
+  }
+
+  const { isOpen } = parsedValue;
+
+  const mutationToUpdateUserFullName = useTanMutation({
+    mutationFn: (values: { email: string; full_name: string }) => {
+      return baiClient.update_full_name(values.email, values.full_name);
+    },
+  });
+
+  const mutationToUpdateUserPassword = useTanMutation({
+    mutationFn: (values: {
+      old_password: string;
+      new_password: string;
+      new_password2: string;
+    }) => {
+      return baiClient.update_password(
+        values.old_password,
+        values.new_password,
+        values.new_password2
+      );
+    },
+  });
+
+  const _onSelectAccessKey = (value: string) => {
+    let matchLoggedAccount = _.find(keyPairInfo.keypairs, [
+      "secret_key",
+      value,
+    ]);
+    form.setFieldsValue({
+      secret_key: matchLoggedAccount.secret_key,
+    });
+  };
+
+  const _updateFullName = (newFullName: string) => {
+    if (full_name !== newFullName) {
+      mutationToUpdateUserFullName.mutate(
+        {
+          full_name: newFullName,
+          email: email,
+        },
+        {
+          onSuccess: () => {
+            _showSuccessMessage(t("webui.menu.FullnameUpdated"));
+            dispatchEvent("updateFullName", { newFullName });
+          },
+          onError: (error: any) => {
+            _showErrorMessage(error.message);
+          },
+        }
+      );
+    }
+  };
+
+  const _updatePassword = (
+    oldPassword: string,
+    newPassword: string,
+    newPassword2: string
+  ) => {
+    if (!oldPassword && !newPassword && !newPassword2) {
+      dispatchEvent("cancel", null);
+      return;
+    }
+    if (!oldPassword) {
+      _showErrorMessage(t("webui.menu.InputOriginalPassword"));
+      return;
+    }
+    if (!newPassword) {
+      _showErrorMessage(t("webui.menu.InputNewPassword"));
+      return;
+    }
+    if (newPassword !== newPassword2) {
+      _showErrorMessage(t("webui.menu.NewPasswordMismatch"));
+      return;
+    }
+    mutationToUpdateUserPassword.mutate(
+      {
+        old_password: oldPassword,
+        new_password: newPassword,
+        new_password2: newPassword2,
+      },
+      {
+        onSuccess: () => {
+          _showSuccessMessage(t("webui.menu.PasswordUpdated"));
+          dispatchEvent("cancel", null);
+        },
+        onError: (error: any) => {
+          _showErrorMessage(error.message);
+        },
+        onSettled: () => {
+          form.setFieldsValue({
+            originalPassword: "",
+            newPassword: "",
+            newPasswordConfirm: "",
+          });
+        },
+      }
+    );
+  };
+
+  const _onSubmit = () => {
```

Comment:
Please remove the `_`` prefix. This component is a function component. All functions defined inside of a function are not accessible outside of the function. Therefore, it is not necessary to prefix functions with an underscore in this case.

However, it may still be needed in some cases. For example, when the component has onSubmit props and you want to override onSubmit inside of the component.